scalyr_agent/connection.py

```diff
@@ -258,7 +258,7 @@ def connect(self):
             self.sock = socket.create_connection((self.host, self.port), self.__timeout)
         else:
             self.sock = create_connection_helper(self.host, self.port, timeout=self.__timeout)
-        if self._tunnel_host:
+        if hasattr(self, '_tunnel_host') and self._tunnel_host:
```

Comment:
here's the first one liner

Reply:
Curious.  I'm still investing to see why this might be happening.